smartsim/_core/_install/buildenv.py

```diff
@@ -0,0 +1,355 @@
+            return Version('.'.join((str(item) for item in vers)))
+        else:
+            raise InvalidVersion(vers)
+
+    @property
+    def major(self):
+        return Version(self).major
+
+    @property
+    def minor(self):
+        return Version(self).minor
+
+    @property
+    def micro(self):
+        return Version(self).micro
+
+    def __gt__(self, cmp):
+        try:
+            return Version(self).__gt__(self._convert_to_version(cmp))
+        except InvalidVersion:
+            return super().__gt__(cmp)
+
+    def __lt__(self, cmp):
+        try:
+            return Version(self).__lt__(self._convert_to_version(cmp))
+        except InvalidVersion:
+            return super().__lt__(cmp)
+
+    def __eq__(self, cmp):
+        try:
+            return Version(self).__eq__(self._convert_to_version(cmp))
+        except InvalidVersion:
+            return super().__eq__(cmp)
+
+    def __ge__(self, cmp):
+        try:
+            return Version(self).__ge__(self._convert_to_version(cmp))
+        except InvalidVersion:
+            return super().__ge__(cmp)
+
+    def __le__(self, cmp):
+        try:
+            return Version(self).__le__(self._convert_to_version(cmp))
+        except InvalidVersion:
+            return super().__le__(cmp)
+
+
+def get_env(var, default):
+    return os.environ.get(var, default)
+
+class RedisAIVersion(Version_):
+
+    defaults = {
+        "1.2.3": {
+            "tensorflow": "2.4.2",
+            "onnx": "1.7.0",
+            "skl2onnx": "1.9.0",
+            "onnxmltools": "1.7.0",
+            "scikit-learn": "0.24.2",
+            "torch": "1.7.1",
+            "torchvision": "0.8.2"
+        },
+        "1.2.4": {
+            "tensorflow": "2.4.2",
+            "onnx": "1.7.0",
+            "skl2onnx": "1.9.0",
+            "onnxmltools": "1.7.0",
+            "scikit-learn": "0.24.2",
+            "torch": "1.7.1",
+            "torchvision": "0.8.2"
+        },
+        "1.2.5": {
+            "tensorflow": "2.6.2",
+            "onnx": "1.9.0",
+            "skl2onnx": "1.10.3",
+            "onnxmltools": "1.10.0",
+            "scikit-learn": "1.0.2",
+            "torch": "1.9.1",
+            "torchvision": "0.10.1"
+        }
+    }
+
+    def __init__(self, vers):
+        if vers not in self.defaults:
+            raise SetupError(
+                f"Invalid RedisAI version {vers}. Options are {self.defaults.keys()}")
+        self.version = vers
+
+    def __getattr__(self, name):
+        return self.defaults[self.version][name]
+
+    def get_defaults(self):
+        return self.defaults[self.version]
+
+
+class Versioner:
+    """Buildtime configuration of third-party dependencies"""
+
+    # compatible Python version
+    PYTHON_MIN = Version_("3.7.0")
+
+    # Versions
+    SMARTSIM = Version_(get_env("SMARTSIM_VERSION", "0.3.2"))
+    SMARTREDIS = Version_(get_env("SMARTREDIS_VERSION", "0.2.0"))
+    SMARTSIM_SUFFIX = get_env("SMARTSIM_SUFFIX", "")
```

Comment:
So a couple things here:
- those default to ``None`` because no value is an acceptable value. For others, no value is not acceptable, we have to have a default.
- For the env flags, unless the user specifically sets something there, we don't want to introduce new flags. again, ``None`` is an acceptable value. 
- These are truthy values meaning that they can all be ``bool``ed. An empty string has the same effect as ``None`` in this case.
-  As a default value, this also indicates to the user/developer that this should be a string. Because we don't use type hinting, a good default should indicate the type if possible.

Reply:
👍